StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -59,26 +59,18 @@ public void Dispose()
             try
             {
                 b_1_0 = new global::B(c: c_1_1);
-                try
-                {
-                    owned_1_0 = new global::StrongInject.Owned<global::B>(b_1_0, () =>
```

Comment:
I did that first, but that resulted in this code which still treats the `Owned` constructor as though it might throw:

```cs
try
{
    b_1_0 = new global::B(c: c_1_1);
    
    owned_1_0 = new global::StrongInject.Owned<global::B>(b_1_0, () =>
    {
        ((global::System.IDisposable)b_1_0).Dispose();
        ((global::System.IDisposable)c_1_1).Dispose();
    });
}
catch
{
    ((global::System.IDisposable)c_1_1).Dispose();
    throw;
}
```

Reply:
Both ways have that advantage, right? The way I prefer has the same number of `try` blocks, but it also has a further advantage in that it doesn't expand `try` blocks with operations that don't need to be inside them.

Reply:
What if an owned is the dependency for something else? (not a func). How would the two look?

Reply:
If it's a direct dependency, it behaves the same both ways:

```cs
public record A(Owned<B> b);
public record B(C c) : IDisposable { public void Dispose() { } }
public record C : IDisposable { public void Dispose() { } }
```

```cs
global::C c_0_1;
global::B b_0_0;
global::StrongInject.Owned<global::B> owned_0_1;
global::A a_0_0;
c_0_1 = new global::C();
try
{
    b_0_0 = new global::B(c: c_0_1);
    try
    {
        owned_0_1 = new global::StrongInject.Owned<global::B>(b_0_0, () =>
        {
            ((global::System.IDisposable)b_0_0).Dispose();
            ((global::System.IDisposable)c_0_1).Dispose();
        });
        a_0_0 = new global::A(b: owned_0_1);
    }
    catch
    {
        ((global::System.IDisposable)b_0_0).Dispose();
        throw;
    }
}
catch
{
    ((global::System.IDisposable)c_0_1).Dispose();
    throw;
}
```

Reply:
> But wont the original way have the advantage that you can remove the try even if it isn't the last operation?

Oh, I just got what you meant. You're right, I think, except I didn't think of a way to test this yet.

Reply:
You'd get the real improvement for:

```csharp
public record A(B b)
public record B(Owned<C> c) : IDisposable { public void Dispose() { } };
public record C(D d) : IDisposable { public void Dispose() { } }
public record D : IDisposable { public void Dispose() { } }
```

where you could eliminate an entire try catch

Reply:
I think you'd find this easier if you removed the try-catch in place, as suggested above. It doesn't particularly bother me that the `Owned` is in the try catch block when it doesn't need to be, since the code will be functionally the same, and so I'd go for the simpler solution.

Reply:
We could also do both... but it will not bother me either.

Reply:
We could but question is whether added code complexity is worth more beautified code. At the moment I'm not convinced.

Reply:
With the local functions rewrite, the no-`try` behavior seems to have fallen out by default.